src/main/lrsql/input/admin/status.clj

```diff
@@ -0,0 +1,51 @@
+(ns lrsql.input.admin.status
+  (:require [clojure.spec.alpha :as s]
+            [com.yetanalytics.squuid :as squuid]
+            [lrsql.spec.admin.status :as stat]
+            [lrsql.util :as u]))
+
+(s/fdef query-timeline-input
+  :args (s/cat :params stat/get-status-params-spec)
+  :ret stat/query-timeline-input-spec)
+
+(def unit-for
+  "Mapping of timeline bucket time unit to suitable FOR sql substring arg."
+  {"year"   4
+   "month"  7
+   "day"    10
+   "hour"   13
+   "minute" 16
+   "second" 19})
+
+(defn query-timeline-input
+  "Transform parameters for timeline into values suitable for query."
+  [{:keys [timeline-unit
+           timeline-since
+           timeline-until]
+    :or   {timeline-unit "day"}}]
+  (let [since' (or (some-> timeline-since u/str->time)
+                   (u/offset-time (u/current-time) -14 :days))
+        until' (or (some-> timeline-until u/str->time)
+                   (u/current-time))]
+    {:unit-for (get unit-for timeline-unit)
+     :since-id (squuid/time->uuid since')
```

Comment:
Just be careful about how the underlying `squuid/time->uuid` works:
```clojure
(time->uuid #inst "2021-12-22T14:33:04.769000000-00:00")
;; => #uuid "017de28f-5801-8fff-8fff-ffffffffffff"
```
Notice how the random digits are all maxed out. Meaning that if you have a randomly generated SQUUID with the same time bits but with random bits less than the maximum, it will fall outside of the range. So you'd want to decrement the `since'` timestamp by a nanosecond/millisecond (whichever is the smallest time interval).

Reply:
> whichever is the smallest time interval

Do you mean whichever is the smallest time interval the Instant API allows?

Reply:
Yeah...now that I recall though, the colossal-squuid library uses milliseconds as the smallest time interval for its purposes, so go with that.

Reply:
https://github.com/yetanalytics/lrsql/pull/282/commits/64d17f39ba3c5f3189eb023c7dd2d6b2c04191b6